File: WitchcraftApp/src/main/java/com/tarbadev/witchcraft/domain/usecase/SetFavoriteRecipeUseCase.java
package com.tarbadev.witchcraft.domain.usecase;

import com.tarbadev.witchcraft.domain.repository.RecipeRepository;
import org.springframework.stereotype.Component;

@Component
public class SetFavoriteRecipeUseCase {
  private final RecipeRepository recipeRepository;

  public SetFavoriteRecipeUseCase(RecipeRepository recipeRepository) {
    this.recipeRepository = recipeRepository;
  }

  public void execute(int id, boolean favorite) {
    recipeRepository.setFavorite(id, favorite);
  }
}

File: WitchcraftApp/src/main/java/com/tarbadev/witchcraft/recipes/domain/usecase/SetFavoriteRecipeUseCase.kt
package com.tarbadev.witchcraft.recipes.domain.usecase

import com.tarbadev.witchcraft.recipes.domain.entity.Recipe
import com.tarbadev.witchcraft.recipes.domain.repository.RecipeRepository
import org.springframework.stereotype.Component

@Component
class SetFavoriteRecipeUseCase(private val recipeRepository: RecipeRepository) {

    fun execute(id: Int, favorite: Boolean): Recipe {
        return recipeRepository.setFavorite(id, favorite)
    }
}
